src/app/nav/navigate-action.js

```diff
@@ -0,0 +1,14 @@
+
+/**
+ * An action creator for in-app page navigation.
+ */
+
+module.exports = function (actionContext, payload, done) {
+  var route = actionContext.router.getRoute(payload.url);
+  if (route) {
+    actionContext.dispatch('navigate', route);
+    done();
+  } else {
+    done({ failed: true });
```

Comment:
Does this establish a convention, or follow one? I'm curious why `{failed: true}` but not `{failed:false}` in the other case.